src/mlpack/core/tree/cellbound_impl.hpp

```diff
@@ -169,34 +169,55 @@ inline void CellBound<MetricType, ElemType>::Center(
 }
 
 template<typename MetricType, typename ElemType>
+template<typename MatType>
 void CellBound<MetricType, ElemType>::AddBound(
     const arma::Col<ElemType>& loCorner,
-    const arma::Col<ElemType>& hiCorner)
+    const arma::Col<ElemType>& hiCorner,
+    const MatType& data)
 {
   assert(numBounds < loBound.n_cols);
   assert(loBound.n_rows == dim);
   assert(loCorner.n_elem == dim);
   assert(hiCorner.n_elem == dim);
 
-  // If the subrectangle is not contained entirely in the outer rectangle,
-  // we shrink it.
   for (size_t k = 0; k < dim; k++)
   {
-    loBound(k, numBounds) = std::max(loCorner[k], bounds[k].Lo());
+    loBound(k, numBounds) = std::numeric_limits<ElemType>::max();
+    hiBound(k, numBounds) = std::numeric_limits<ElemType>::lowest();
+  }
 
-    hiBound(k, numBounds) = std::min(bounds[k].Hi(), hiCorner[k]);
+  for (size_t i = 0; i < data.n_cols; i++)
```

Comment:
Indeed, that provides some benefits. If I am not mistaken, the minimum bounding rectangle is used in the `Contains()` function but I can replace that by the hyperrectangles.